src/platforms/dotnet/guides/aspnet/performance/included-instrumentation.mdx

```diff
@@ -0,0 +1,142 @@
+---
+title: Included Instrumentation
+sidebar_order: 20
+description: "Learn what transactions are captured after tracing is enabled."
+---
+
+<Note>
+
+Capturing transactions requires that you first <PlatformLink to="/performance/">set up performance monitoring</PlatformLink> if you haven't already. You'll also need to enable the <PlatformLink to="/">base Sentry integration</PlatformLink>.
+
+</Note>
+
+ASP.NET integration, once enabled, captures each incoming HTTP request and turns it into a transaction with the help of a custom middleware. To enable tracing on ASP.NET, you need to update your `Global.asax.cs` file like this:
+
+```csharp
+using System;
+using System.Configuration;
+using System.Web.Mvc;
+using System.Web.Routing;
+using Sentry;
+using Sentry.AspNet;
+
+namespace AspNetMvc5Ef6
+{
+    public class MvcApplication : System.Web.HttpApplication
+    {
+        private IDisposable _sentry;
+
+        protected void Application_Start()
+        {
+            // ...
+
+            // Set up the sentry SDK
+            _sentry = SentrySdk.Init(o =>
+            {
+                // We store the DSN inside Web.config; make sure to use your own DSN!
+                o.Dsn = ConfigurationManager.AppSettings["SentryDsn"];
+
+                // Get ASP.NET integration
+                o.AddAspNet();
+
+                // Collect 100% traces for testing purposes
+                // (consider using a smaller value when running in production)
+                o.TracesSampleRate = 1.0;
+            });
+        }
+
+        // Global error catcher
+        protected void Application_Error()
+        {
+            var exception = Server.GetLastError();
+
+            // Capture unhandled exceptions
+            SentrySdk.CaptureException(exception);
+        }
+
+        protected void Application_End()
+        {
+            // Close the Sentry SDK (flushes queued events to Sentry)
+            _sentry?.Dispose();
+        }
+
+        protected void Application_BeginRequest()
+        {
+            // Start a transaction that encompasses the current request
+            var transaction = Context.StartSentryTransaction();
+
+            // Attach transaction to the request context to finish it when the request ends
+            Context.Items["__SentryTransaction"] = transaction;
+        }
+
+        protected void Application_EndRequest()
+        {
+            // Finish the currently active transaction
+            if (Context.Items.Contains("__SentryTransaction"))
```

Comment:
Happy to merge but looks like we found a second extensions method here to be added eh?

`            Context.Items["__SentryTransaction"] = transaction;` can go into `StartSentryTransaction`

Reply:
I don't know if it should because it escapes the boundaries of the method. I think the user should be in control of what gets put into `Context.Items`.

Reply:
Most users don't care if you add `___SentryBla` in there and we need to optimize for ease of use.
First user that complains we can add an opt-out but I doubt this will ever happen.

Right now there's still a LOT of boiler plate for something that should be automatic

Reply:
Then should we also add another method called `Context.FinishSentryTransaction()`?

Reply:
That would be awesome I think. It's opinionated but super simple for the user.

Reply:
See https://github.com/getsentry/sentry-dotnet/pull/1011